addons/mail/static/src/core/common/composer.js

```diff
@@ -748,23 +752,32 @@ export class Composer extends Component {
 
     async editMessage() {
         const composer = toRaw(this.props.composer);
-        if (composer.text || composer.message.attachment_ids.length > 0) {
+        if (this.hasContentToUpdate) {
             await this.processMessage(async (value) =>
                 composer.message.edit(value, composer.attachments, {
                     mentionedChannels: composer.mentionedChannels,
                     mentionedPartners: composer.mentionedPartners,
                 })
             );
         } else {
-            this.env.services.dialog.add(MessageConfirmDialog, {
-                message: composer.message,
-                onConfirm: () => this.message.remove(),
-                prompt: _t("Are you sure you want to delete this message?"),
-            });
+            this.env.services.dialog.add(
+                MessageConfirmDialog,
+                {
+                    message: composer.message,
+                    onConfirm: () => this.message.remove(),
+                    prompt: _t("Are you sure you want to delete this message?"),
+                },
+                { context: this }
```

Comment:
To make sure this dialog is opened in the overlay of its own root in case of having several overlays at the same time.

Reply:
How do we get several overlays at the same time in this case?

Reply:
LC App is mounted using the same env than the light DOM. Calling `dialog.add` will add an overlay to the `overlay_service`, which will result in 1 overlay in the light DOM and one in the shadow DOM.